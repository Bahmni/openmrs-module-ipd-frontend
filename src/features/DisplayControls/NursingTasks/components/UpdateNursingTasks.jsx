import React, { useEffect, useState } from "react";
import { FormattedMessage } from "react-intl";
import PropTypes from "prop-types";
import "../styles/UpdateNursingTasks.scss";
import SideBarPanel from "../../../SideBarPanel/components/SideBarPanel";
import SaveAndCloseButtons from "../../../SaveAndCloseButtons/components/SaveAndCloseButtons";
import Clock from "../../../../icons/clock.svg";
import {
  Modal,
  OverflowMenu,
  OverflowMenuItem,
  TextArea,
  Toggle,
} from "carbon-components-react";
import moment from "moment";
import { TimePicker24Hour, Title } from "bahmni-carbon-ui";
import AdministeredMedicationList from "./AdministeredMedicationList";
import {
  saveAdministeredMedication,
  isTimeWithinAdministeredWindow,
} from "../utils/NursingTasksUtils";
import { SideBarPanelClose } from "../../../SideBarPanel/components/SideBarPanelClose";
import { performerFunction } from "../utils/constants";
import DisplayTags from "../../../../components/DisplayTags/DisplayTags";

const UpdateNursingTasks = (props) => {
  const {
    medicationTasks,
    updateNursingTasksSlider,
    patientId,
    providerId,
    setShowSuccessNotification,
  } = props;
  const [tasks, updateTasks] = useState({});
  const [errors, updateErrors] = useState({});
  const [showErrors, updateShowErrors] = useState(false);
  const [isSaveDisabled, updateIsSaveDisabled] = useState(true);
  const [isAnyMedicationSkipped, setIsAnyMedicationSkipped] = useState(false);
  const [openConfirmationModal, setOpenConfirmationModal] = useState(false);
  const [isAnyMedicationAdministered, setIsAnyMedicationAdministered] =
    useState(false);
  const [administeredTasks, setAdministeredTasks] = useState({});
  const [skippedTasks, setSkippedTasks] = useState({});
  const [showWarningNotification, setShowWarningNotification] = useState(false);
  const invalidTimeText24Hour = (
    <FormattedMessage
      id={"INVALID_TIME"}
      defaultMessage={"Please enter valid time"}
    />
  );
  const closeModal = () => {
    setOpenConfirmationModal(false);
  };

  const saveAdministeredTasks = () => {
    setShowSuccessNotification(true);
    setOpenConfirmationModal(false);
    updateNursingTasksSlider(false);
  };

  const handlePrimaryButtonClick = async () => {
    const administeredTasks = createAdministeredTasksPayload();
    const response = await saveAdministeredMedication(administeredTasks);
    response.status === 200 ? saveAdministeredTasks() : null;
  };

  const createAdministeredTasksPayload = () => {
    const administeredTasksPayload = [];
    Object.keys(administeredTasks).forEach((key) => {
      const time = new Date(tasks[key].actualTime);
      const utcTimeEpoch = moment.utc(time).unix();

      administeredTasksPayload.push({
        patientUuid: patientId,
        orderUuid: administeredTasks[key].orderId,
        providers: [{ providerUuid: providerId, function: performerFunction }],
        notes: administeredTasks[key].notes
          ? [{ authorUuid: providerId, text: administeredTasks[key]?.notes }]
          : [],
        status: administeredTasks[key]?.status,
        slotUuid: key,
        administeredDateTime: utcTimeEpoch,
      });
    });
    Object.keys(skippedTasks).forEach((key) => {
      administeredTasksPayload.push({
        patientUuid: patientId,
        orderUuid: skippedTasks[key].orderId,
        providers: [{ providerUuid: providerId, function: performerFunction }],
        notes: skippedTasks[key].notes
          ? [{ authorUuid: providerId, text: administeredTasks[key]?.notes }]
          : [],
        status: skippedTasks[key]?.status,
        slotUuid: key,
      });
    });
    return administeredTasksPayload;
  };

  const getLabel = (time) => {
    return (
      <div>
        <FormattedMessage id={"DONE"} defaultMessage={"Done"} />
        &nbsp;
        {time && (
          <>
            <FormattedMessage id={"AT"} defaultMessage={"at"} />
            &nbsp;{time.format("HH:mm")}
          </>
        )}
      </div>
    );
  };
  useEffect(() => {
    medicationTasks.map((medicationTask) => {
      updateTasks((prev) => {
        return {
          ...prev,
          [medicationTask.uuid]: {
            displayName: medicationTask.drugName,
            doseType: medicationTask.doseType,
            dosage: medicationTask.dosage,
            route: medicationTask.drugRoute,
            startTime: medicationTask.startTime,
            dosingInstructions: medicationTask.dosingInstructions,
            isSelected: false,
            actualTime: null,
            providerId: medicationTask.providerId,
            orderId: medicationTask.orderId,
          },
        };
      });
    });
  }, [medicationTasks]);

  useEffect(() => {
    checkFormStatus();
  }, [tasks]);

  const checkFormStatus = () => {
    let saveDisabled = true;
    Object.keys(tasks).forEach((key) => {
      if (tasks[key].isSelected) {
        saveDisabled = false;
        setIsAnyMedicationAdministered(true);
      }
      if (tasks[key].skipped) {
        saveDisabled = false;
        setIsAnyMedicationSkipped(true);
      }
    });
    updateIsSaveDisabled(saveDisabled);
  };

  const handleTimeChange = (time, id) => {
    if (!isTimeWithinAdministeredWindow(time, tasks[id].startTime)) {
      updateTasks({
        ...tasks,
        [id]: {
          ...tasks[id],
          isTimeOutOfWindow: true,
          actualTime: moment(time, "HH:mm"),
        },
      });
      updateErrors({
        ...errors,
        [id]: Boolean(!tasks[id].notes),
      });
    } else {
      updateTasks({
        ...tasks,
        [id]: {
          ...tasks[id],
          isTimeOutOfWindow: false,
          actualTime: moment(time, "HH:mm"),
        },
      });
      delete errors[id];
    }
  };

  const handleToggle = (checked, id) => {
    const time = moment().format("HH:mm");
    if (checked && !isTimeWithinAdministeredWindow(time, tasks[id].startTime)) {
      updateTasks({
        ...tasks,
        [id]: {
          ...tasks[id],
          isTimeOutOfWindow: true,
          isSelected: checked,
          actualTime: checked ? moment() : null,
        },
      });
      updateErrors({
        ...errors,
        [id]: Boolean(!tasks[id].notes),
      });
    } else {
      updateTasks({
        ...tasks,
        [id]: {
          ...tasks[id],
          isTimeOutOfWindow: false,
          isSelected: checked,
          actualTime: checked ? moment() : null,
        },
      });
      delete errors[id];
    }
  };

  const handleNotes = (e, id) => {
    updateTasks({
      ...tasks,
      [id]: {
        ...tasks[id],
        notes: e.target.value,
      },
    });
    if (e.target.value) {
      delete errors[id];
    } else {
      if (tasks[id].isTimeOutOfWindow || tasks[id].skipped) {
        updateErrors({
          ...errors,
          [id]: true,
        });
      }
    }
  };

  const handleSave = () => {
    if (Object.keys(errors).length === 0) {
      setOpenConfirmationModal(true);
    }

    updateShowErrors(true);

    setTimeout(() => {
      updateShowErrors(false);
    }, 3000);

    setAdministeredTasks({});
    setSkippedTasks({});
    Object.keys(tasks).forEach((key) => {
      if (tasks[key].isSelected) {
        setAdministeredTasks((prev) => ({
          ...prev,
          [key]: { ...tasks[key], status: "completed" },
        }));
      } else if (tasks[key].skipped) {
        setSkippedTasks((prev) => ({
          ...prev,
          [key]: { ...tasks[key], status: "not-done" },
        }));
      }
    });
  };

  const handleClose = () => {
    if (isSaveDisabled) {
      updateNursingTasksSlider(false);
      setShowWarningNotification(false);
    } else {
      setShowWarningNotification(true);
    }
  };

  const handleSkipDrug = (medicationTask, skipped) => {
    updateTasks({
      ...tasks,
      [medicationTask.uuid]: {
        ...tasks[medicationTask.uuid],
        skipped: skipped,
      },
    });
    if (skipped && !tasks[medicationTask.uuid].notes) {
      updateErrors({
        ...errors,
        [medicationTask.uuid]: skipped,
      });
    } else {
      delete errors[medicationTask.uuid];
    }
  };

  const sliderCloseActions = {
    onCancel: () => {
      setShowWarningNotification(false);
      updateNursingTasksSlider(false);
    },
    onClose: () => {
      setShowWarningNotification(false);
    },
  };

  return (
    <>
      <SideBarPanel
        title={<FormattedMessage id="TASKS" defaultMessage={"Task(s)"} />}
        closeSideBar={handleClose}
      >
        <div className={"update-nursing-tasks"}>
          <div className={"task-heading"}>
            <FormattedMessage
              id={"SCHEDULED_FOR_KEY"}
              defaultMessage={"Scheduled for"}
            />
            <Clock />
            {medicationTasks[0].startTime}
          </div>
          {medicationTasks.map((medicationTask, index) => {
            return (
              <div key={index} className={"nursing-task-section"}>
                {!tasks[medicationTask.uuid]?.skipped && (
                  <Toggle
                    id={medicationTask.uuid}
                    size={"sm"}
                    labelA={getLabel(tasks[medicationTask.uuid]?.actualTime)}
                    labelB={getLabel(tasks[medicationTask.uuid]?.actualTime)}
                    onToggle={handleToggle}
                  />
                )}
                <div className={"medication-name"}>
                  <div
                    className={`name ${
                      tasks[medicationTask.uuid]?.skipped && "red-text"
                    }`}
                  >
                    {medicationTask.drugName}
                  </div>
                  <DisplayTags drugOrder={medicationTask.dosingInstructions} />
                </div>
                <div className="medication-details">
                  <span>{medicationTask.dosage}</span>
                  {medicationTask.doseType && (
                    <span>&nbsp;-&nbsp;{medicationTask.doseType}</span>
                  )}
                  <span>&nbsp;-&nbsp;{medicationTask.drugRoute}</span>
                </div>
                {(tasks[medicationTask.uuid]?.actualTime ||
                  tasks[medicationTask.uuid]?.skipped) && (
                  <div style={{ display: "flex" }}>
                    {tasks[medicationTask.uuid]?.actualTime && (
                      <TimePicker24Hour
                        defaultTime={tasks[
                          medicationTask.uuid
                        ]?.actualTime.format("HH:mm")}
                        onChange={(time) => {
                          handleTimeChange(time, medicationTask.uuid);
                        }}
                        labelText="Task Time"
                        invalidText={invalidTimeText24Hour}
                        light={true}
                      />
                    )}
                    <div
                      className={`${
                        Boolean(tasks[medicationTask.uuid]?.actualTime) &&
                        "notes-text-area"
                      }`}
                      style={{ width: "100%" }}
                    >
                      <TextArea
                        labelText={
                          <Title
                            text={"Notes"}
                            isRequired={
                              tasks[medicationTask.uuid].isTimeOutOfWindow ||
                              tasks[medicationTask.uuid].skipped
                            }
                          />
                        }
                        onChange={(e) => {
                          handleNotes(e, medicationTask.uuid);
                        }}
                        maxCount={250}
                        rows={1}
                        cols={50}
                        light={true}
                      />
                      {showErrors && errors[medicationTask.uuid] && (
                        <div className={"error"}>
                          <FormattedMessage
                            id={"NOTES_ERROR_MESSAGE"}
                            defaultMessage={"Please enter notes"}
                          />
                        </div>
                      )}
                    </div>
                  </div>
                )}
                <OverflowMenu
                  flipped={true}
                  disabled={tasks[medicationTask.uuid]?.isSelected}
                  className={"overflowMenu"}
                >
                  {tasks[medicationTask.uuid]?.skipped ? (
                    <OverflowMenuItem
                      itemText={"Un-Skip Drug"}
                      onClick={() => {
                        handleSkipDrug(medicationTask, false);
                      }}
                    />
                  ) : (
                    <OverflowMenuItem
                      itemText={"Skip Drug"}
                      onClick={() => {
                        handleSkipDrug(medicationTask, true);
                      }}
                    />
                  )}
                </OverflowMenu>
              </div>
            );
          })}
        </div>
        <Modal
          open={
            (isAnyMedicationAdministered || isAnyMedicationSkipped) &&
            openConfirmationModal
          }
          onRequestClose={closeModal}
          onSecondarySubmit={closeModal}
          preventCloseOnClickOutside={true}
          modalHeading={
            <FormattedMessage
              id={"NURSING_TASKS_CONFIRMATION"}
              defaultMessage={"Please confirm your nursing tasks"}
            />
          }
          primaryButtonText={
            <FormattedMessage
              id={"DRUG_CHART_MODAL_SAVE"}
              defaultMessage={"Save"}
            />
          }
          secondaryButtonText={
            <FormattedMessage
              id={"DRUG_CHART_MODAL_CANCEL"}
              defaultMessage={"Cancel"}
            />
          }
          onRequestSubmit={handlePrimaryButtonClick}
        >
          <hr />
          <AdministeredMedicationList
            list={{ ...administeredTasks, ...skippedTasks }}
          />
        </Modal>
        <SaveAndCloseButtons
          onSave={handleSave}
          onClose={handleClose}
          isSaveDisabled={isSaveDisabled}
        />
      </SideBarPanel>

      {showWarningNotification && (
        <SideBarPanelClose
          className="warning-notification"
          open={true}
          message={
            <FormattedMessage
              id="TREATMENTS_WARNING_TEXT"
              defaultMessage="You will lose the details entered. Do you want to continue?"
            />
          }
          label={""}
          primaryButtonText={<FormattedMessage id="NO" defaultMessage="No" />}
          secondaryButtonText={
            <FormattedMessage id="YES" defaultMessage="Yes" />
          }
          onSubmit={sliderCloseActions.onClose}
          onSecondarySubmit={sliderCloseActions.onCancel}
          onClose={sliderCloseActions.onClose}
        />
      )}
    </>
  );
};

UpdateNursingTasks.propTypes = {
  medicationTasks: PropTypes.array.isRequired,
  updateNursingTasksSlider: PropTypes.func.isRequired,
  patientId: PropTypes.string.isRequired,
  providerId: PropTypes.string.isRequired,
  setShowSuccessNotification: PropTypes.func.isRequired,
};
export default UpdateNursingTasks;
