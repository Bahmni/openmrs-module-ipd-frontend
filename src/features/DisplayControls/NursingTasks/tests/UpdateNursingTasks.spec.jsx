import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import UpdateNursingTasks from "../components/UpdateNursingTasks";
import { mockMedicationTasks } from "./NursingTasksUtilsMockData";

describe("UpdateNursingTasksSlider", function () {
  it("should render UpdateNursingTasksSlider", function () {
    const { container } = render(
      <UpdateNursingTasks
        medicationTasks={mockMedicationTasks}
        updateNursingTasksSlider={jest.fn}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should enable save Button when atleast one task is selected", function () {
    const { container } = render(
      <UpdateNursingTasks
        medicationTasks={mockMedicationTasks}
        updateNursingTasksSlider={jest.fn}
      />
    );
    const saveButton = screen.getByText("Save");
    expect(saveButton.disabled).toEqual(true);
    const toggleButton = container.querySelectorAll(".bx--toggle__switch")[0];
    fireEvent.click(toggleButton);
    expect(saveButton.disabled).toEqual(false);
  });

  it("should show notes and time when toggle switch is On", function () {
    const { container } = render(
      <UpdateNursingTasks
        medicationTasks={mockMedicationTasks}
        updateNursingTasksSlider={jest.fn}
      />
    );
    const toggleButton = container.querySelectorAll(".bx--toggle__switch")[0];
    fireEvent.click(toggleButton);

    const timePicker = container.querySelectorAll(
      ".bx--time-picker__input-field"
    )[0];
    expect(timePicker).toBeTruthy();
    const notes = container.querySelectorAll(".bx--text-area")[0];
    expect(notes).toBeTruthy();
  });

  it("should show warning for empty notes when time is updated", function () {
    const { container } = render(
      <UpdateNursingTasks
        medicationTasks={mockMedicationTasks}
        updateNursingTasksSlider={jest.fn}
      />
    );
    const toggleButton = container.querySelectorAll(".bx--toggle__switch")[0];
    fireEvent.click(toggleButton);

    const timePicker = screen.getAllByRole("textbox")[0];
    fireEvent.change(timePicker, { target: { value: "12:00" } });
    fireEvent.blur(timePicker);
    const saveButton = screen.getByText("Save");
    fireEvent.click(saveButton);
    expect(screen.getByText("Please enter notes")).toBeTruthy();
  });
});
