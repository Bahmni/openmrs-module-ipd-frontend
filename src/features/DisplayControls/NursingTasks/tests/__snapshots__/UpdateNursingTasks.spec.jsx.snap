// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateNursingTasksSlider should close the slider on click of cancel button when no changes are made 1`] = `
<div>
  <div
    class="side-bar-nav"
  >
    <div
      class="side-bar-header"
    >
      <div>
        Task(s)
      </div>
      <div
        class="close-icon"
      >
        <button
          class="bx--btn bx--btn--ghost bx--tooltip--hidden bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--btn--icon-only--top bx--tooltip--align-center"
          tabindex="0"
          type="button"
        >
          <div
            class="bx--assistive-text"
          >
            Close
          </div>
          <svg
            aria-hidden="true"
            aria-label="Close"
            class="bx--btn__icon"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 32 32"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="side-bar-children"
    >
      <div
        class="update-nursing-tasks"
      >
        <div
          class="task-heading"
        >
          Scheduled for
          <test-file-stub />
          08:00
        </div>
        <div
          class="nursing-task-section"
        >
          <div
            class="bx--form-item"
          >
            <input
              class="bx--toggle-input bx--toggle-input--small"
              id="d9ae1992-63ac-4e05-8dfb-13f021102ec6"
              type="checkbox"
            />
            <label
              aria-label="Toggle"
              class="bx--toggle-input__label"
              for="d9ae1992-63ac-4e05-8dfb-13f021102ec6"
            >
              <span
                class="bx--toggle__switch"
              >
                <svg
                  class="bx--toggle__check"
                  height="5px"
                  viewBox="0 0 6 5"
                  width="6px"
                >
                  <path
                    d="M2.2 2.7L5 0 6 1 2.2 5 0 2.7 1 1.5z"
                  />
                </svg>
                <span
                  aria-hidden="true"
                  class="bx--toggle__text--off"
                >
                  <div>
                    Done
                     
                  </div>
                </span>
                <span
                  aria-hidden="true"
                  class="bx--toggle__text--on"
                >
                  <div>
                    Done
                     
                  </div>
                </span>
              </span>
            </label>
          </div>
          <div
            class="medication-name"
          >
            <div
              class="name undefined"
            >
              Amoxicillin/Clavulanic Acid  125 mg/31.25 mg/5 mL Powder for Oral Suspension
            </div>
            <div
              class="bx--tag bx--tag--blue"
              id="tag-49"
            >
              
              <span
                title="Rx"
              >
                Rx
              </span>
            </div>
          </div>
          <div
            class="medication-details"
          >
            <span>
              30mg
            </span>
            <span>
               - 
              Topical
            </span>
          </div>
          <button
            aria-expanded="false"
            aria-haspopup="true"
            aria-label="Open and close list of options"
            class="overflowMenu bx--overflow-menu bx--overflow-menu--md"
            type="button"
          >
            <svg
              aria-label="Open and close list of options"
              class="bx--overflow-menu__icon"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="16"
                cy="8"
                r="2"
              />
              <circle
                cx="16"
                cy="16"
                r="2"
              />
              <circle
                cx="16"
                cy="24"
                r="2"
              />
              <title>
                Open and close list of options
              </title>
            </svg>
          </button>
        </div>
        <div
          class="nursing-task-section"
        >
          <div
            class="bx--form-item"
          >
            <input
              class="bx--toggle-input bx--toggle-input--small"
              id="b9904678-eb07-4eb0-8d03-53fcaa1d20d0"
              type="checkbox"
            />
            <label
              aria-label="Toggle"
              class="bx--toggle-input__label"
              for="b9904678-eb07-4eb0-8d03-53fcaa1d20d0"
            >
              <span
                class="bx--toggle__switch"
              >
                <svg
                  class="bx--toggle__check"
                  height="5px"
                  viewBox="0 0 6 5"
                  width="6px"
                >
                  <path
                    d="M2.2 2.7L5 0 6 1 2.2 5 0 2.7 1 1.5z"
                  />
                </svg>
                <span
                  aria-hidden="true"
                  class="bx--toggle__text--off"
                >
                  <div>
                    Done
                     
                  </div>
                </span>
                <span
                  aria-hidden="true"
                  class="bx--toggle__text--on"
                >
                  <div>
                    Done
                     
                  </div>
                </span>
              </span>
            </label>
          </div>
          <div
            class="medication-name"
          >
            <div
              class="name undefined"
            >
              Sodium Chloride 0.9% (1 L) Infusion Bag (Normal Saline)
            </div>
            <div
              class="bx--tag bx--tag--blue"
              id="tag-50"
            >
              
              <span
                title="Rx"
              >
                Rx
              </span>
            </div>
          </div>
          <div
            class="medication-details"
          >
            <span>
              25mg
            </span>
            <span>
               - 
              Intravenous
            </span>
          </div>
          <button
            aria-expanded="false"
            aria-haspopup="true"
            aria-label="Open and close list of options"
            class="overflowMenu bx--overflow-menu bx--overflow-menu--md"
            type="button"
          >
            <svg
              aria-label="Open and close list of options"
              class="bx--overflow-menu__icon"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="16"
                cy="8"
                r="2"
              />
              <circle
                cx="16"
                cy="16"
                r="2"
              />
              <circle
                cx="16"
                cy="24"
                r="2"
              />
              <title>
                Open and close list of options
              </title>
            </svg>
          </button>
        </div>
      </div>
      <div
        class="bx--modal bx--modal-tall"
        role="presentation"
      >
        <span
          class="bx--visually-hidden"
          role="link"
          tabindex="0"
        >
          Focus sentinel
        </span>
        <div
          aria-label="[object Object]"
          aria-modal="true"
          class="bx--modal-container"
          role="dialog"
          tabindex="-1"
        >
          <div
            class="bx--modal-header"
          >
            
            <h3
              class="bx--modal-header__heading"
              id="bx--modal-header__heading--modal-6"
            >
              Please confirm your nursing tasks
            </h3>
            <button
              aria-label="close"
              class="bx--modal-close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="bx--modal-close__icon"
                fill="currentColor"
                focusable="false"
                height="20"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
                />
              </svg>
            </button>
          </div>
          <div
            aria-labelledby="bx--modal-header__heading--modal-6"
            class="bx--modal-content"
            id="bx--modal-body--modal-6"
          >
            <hr />
            <div
              aria-label="Structured list section"
              class="bx--structured-list administered-list-body"
              role="table"
            >
              <div
                class="bx--structured-list-tbody"
                role="rowgroup"
              />
            </div>
          </div>
          <div
            class="bx--modal-footer bx--btn-set"
          >
            <button
              class="bx--btn bx--btn--secondary"
              tabindex="0"
              type="button"
            >
              Cancel
            </button>
            <button
              class="bx--btn bx--btn--primary"
              tabindex="0"
              type="button"
            >
              Save
            </button>
          </div>
        </div>
        <span
          class="bx--visually-hidden"
          role="link"
          tabindex="0"
        >
          Focus sentinel
        </span>
      </div>
      <div
        class="footer"
      >
        <button
          class="cancel-button-slider bx--btn bx--btn--secondary"
          data-testid="cancel"
          tabindex="0"
          type="button"
        >
          <span>
            Cancel
          </span>
        </button>
        <button
          class="save-button-slider bx--btn bx--btn--primary bx--btn--disabled"
          disabled=""
          tabindex="0"
          type="button"
        >
          Save
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`UpdateNursingTasksSlider should render UpdateNursingTasksSlider 1`] = `
<div>
  <div
    class="side-bar-nav"
  >
    <div
      class="side-bar-header"
    >
      <div>
        Task(s)
      </div>
      <div
        class="close-icon"
      >
        <button
          class="bx--btn bx--btn--ghost bx--tooltip--hidden bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--btn--icon-only--top bx--tooltip--align-center"
          tabindex="0"
          type="button"
        >
          <div
            class="bx--assistive-text"
          >
            Close
          </div>
          <svg
            aria-hidden="true"
            aria-label="Close"
            class="bx--btn__icon"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 32 32"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="side-bar-children"
    >
      <div
        class="update-nursing-tasks"
      >
        <div
          class="task-heading"
        >
          Scheduled for
          <test-file-stub />
          08:00
        </div>
        <div
          class="nursing-task-section"
        >
          <div
            class="bx--form-item"
          >
            <input
              class="bx--toggle-input bx--toggle-input--small"
              id="d9ae1992-63ac-4e05-8dfb-13f021102ec6"
              type="checkbox"
            />
            <label
              aria-label="Toggle"
              class="bx--toggle-input__label"
              for="d9ae1992-63ac-4e05-8dfb-13f021102ec6"
            >
              <span
                class="bx--toggle__switch"
              >
                <svg
                  class="bx--toggle__check"
                  height="5px"
                  viewBox="0 0 6 5"
                  width="6px"
                >
                  <path
                    d="M2.2 2.7L5 0 6 1 2.2 5 0 2.7 1 1.5z"
                  />
                </svg>
                <span
                  aria-hidden="true"
                  class="bx--toggle__text--off"
                >
                  <div>
                    Done
                     
                  </div>
                </span>
                <span
                  aria-hidden="true"
                  class="bx--toggle__text--on"
                >
                  <div>
                    Done
                     
                  </div>
                </span>
              </span>
            </label>
          </div>
          <div
            class="medication-name"
          >
            <div
              class="name undefined"
            >
              Amoxicillin/Clavulanic Acid  125 mg/31.25 mg/5 mL Powder for Oral Suspension
            </div>
            <div
              class="bx--tag bx--tag--blue"
              id="tag-3"
            >
              
              <span
                title="Rx"
              >
                Rx
              </span>
            </div>
          </div>
          <div
            class="medication-details"
          >
            <span>
              30mg
            </span>
            <span>
               - 
              Topical
            </span>
          </div>
          <button
            aria-expanded="false"
            aria-haspopup="true"
            aria-label="Open and close list of options"
            class="overflowMenu bx--overflow-menu bx--overflow-menu--md"
            type="button"
          >
            <svg
              aria-label="Open and close list of options"
              class="bx--overflow-menu__icon"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="16"
                cy="8"
                r="2"
              />
              <circle
                cx="16"
                cy="16"
                r="2"
              />
              <circle
                cx="16"
                cy="24"
                r="2"
              />
              <title>
                Open and close list of options
              </title>
            </svg>
          </button>
        </div>
        <div
          class="nursing-task-section"
        >
          <div
            class="bx--form-item"
          >
            <input
              class="bx--toggle-input bx--toggle-input--small"
              id="b9904678-eb07-4eb0-8d03-53fcaa1d20d0"
              type="checkbox"
            />
            <label
              aria-label="Toggle"
              class="bx--toggle-input__label"
              for="b9904678-eb07-4eb0-8d03-53fcaa1d20d0"
            >
              <span
                class="bx--toggle__switch"
              >
                <svg
                  class="bx--toggle__check"
                  height="5px"
                  viewBox="0 0 6 5"
                  width="6px"
                >
                  <path
                    d="M2.2 2.7L5 0 6 1 2.2 5 0 2.7 1 1.5z"
                  />
                </svg>
                <span
                  aria-hidden="true"
                  class="bx--toggle__text--off"
                >
                  <div>
                    Done
                     
                  </div>
                </span>
                <span
                  aria-hidden="true"
                  class="bx--toggle__text--on"
                >
                  <div>
                    Done
                     
                  </div>
                </span>
              </span>
            </label>
          </div>
          <div
            class="medication-name"
          >
            <div
              class="name undefined"
            >
              Sodium Chloride 0.9% (1 L) Infusion Bag (Normal Saline)
            </div>
            <div
              class="bx--tag bx--tag--blue"
              id="tag-4"
            >
              
              <span
                title="Rx"
              >
                Rx
              </span>
            </div>
          </div>
          <div
            class="medication-details"
          >
            <span>
              25mg
            </span>
            <span>
               - 
              Intravenous
            </span>
          </div>
          <button
            aria-expanded="false"
            aria-haspopup="true"
            aria-label="Open and close list of options"
            class="overflowMenu bx--overflow-menu bx--overflow-menu--md"
            type="button"
          >
            <svg
              aria-label="Open and close list of options"
              class="bx--overflow-menu__icon"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="16"
                cy="8"
                r="2"
              />
              <circle
                cx="16"
                cy="16"
                r="2"
              />
              <circle
                cx="16"
                cy="24"
                r="2"
              />
              <title>
                Open and close list of options
              </title>
            </svg>
          </button>
        </div>
      </div>
      <div
        class="bx--modal bx--modal-tall"
        role="presentation"
      >
        <span
          class="bx--visually-hidden"
          role="link"
          tabindex="0"
        >
          Focus sentinel
        </span>
        <div
          aria-label="[object Object]"
          aria-modal="true"
          class="bx--modal-container"
          role="dialog"
          tabindex="-1"
        >
          <div
            class="bx--modal-header"
          >
            
            <h3
              class="bx--modal-header__heading"
              id="bx--modal-header__heading--modal-1"
            >
              Please confirm your nursing tasks
            </h3>
            <button
              aria-label="close"
              class="bx--modal-close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="bx--modal-close__icon"
                fill="currentColor"
                focusable="false"
                height="20"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
                />
              </svg>
            </button>
          </div>
          <div
            aria-labelledby="bx--modal-header__heading--modal-1"
            class="bx--modal-content"
            id="bx--modal-body--modal-1"
          >
            <hr />
            <div
              aria-label="Structured list section"
              class="bx--structured-list administered-list-body"
              role="table"
            >
              <div
                class="bx--structured-list-tbody"
                role="rowgroup"
              />
            </div>
          </div>
          <div
            class="bx--modal-footer bx--btn-set"
          >
            <button
              class="bx--btn bx--btn--secondary"
              tabindex="0"
              type="button"
            >
              Cancel
            </button>
            <button
              class="bx--btn bx--btn--primary"
              tabindex="0"
              type="button"
            >
              Save
            </button>
          </div>
        </div>
        <span
          class="bx--visually-hidden"
          role="link"
          tabindex="0"
        >
          Focus sentinel
        </span>
      </div>
      <div
        class="footer"
      >
        <button
          class="cancel-button-slider bx--btn bx--btn--secondary"
          data-testid="cancel"
          tabindex="0"
          type="button"
        >
          <span>
            Cancel
          </span>
        </button>
        <button
          class="save-button-slider bx--btn bx--btn--primary bx--btn--disabled"
          disabled=""
          tabindex="0"
          type="button"
        >
          Save
        </button>
      </div>
    </div>
  </div>
</div>
`;
